itest/loadtest/load_test.go

```diff
@@ -20,11 +21,30 @@ var (
 		},
 		[]string{"test_name"},
 	)
+
+	memAlloc = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "memory_alloc",
+			Help: "Memory usage of the test execution, in bytes",
+		},
+		[]string{"test_name"},
+	)
+
+	memTotalAlloc = prometheus.NewGaugeVec(
```

Comment:
Do we need this given the prom metrics already export memory related to the test? https://github.com/danielfm/prometheus-for-developers/blob/master/README.md#measuring-memorycpu-usage

Reply:
Instead, I think we need to focus on metrics including (some of this is a matter of enableing the gRPC prom metrics, and utilizing what we've already exported w/ new queries) :
  * time to execute the various components of the test (list asset, coin selection, etc) 
    * this may require deeper telemetry within tapd itself, depending on the way the load tests are written 
  * proof size growth as a result of test instance (size after compared to size before) 
  * total db size (for both sqlite and postgres) 
  * total number of assets created
  * total number of assets sent 
  * 


cc @calvinrzachman as he may already have some of this posted in the dashboard

Reply:
> Do we need this given the prom metrics already export memory related to the test? https://github.com/danielfm/prometheus-for-developers/blob/master/README.md#measuring-memorycpu-usage

Probably not, will take a look at this

Reply:
> time to execute the various components of the test (list asset, coin selection, etc)
this may require deeper telemetry within tapd itself, depending on the way the load tests are written

I think it's better to fully rely on the long-running tapd prometheus metrics that were updated on [tapd/pull/716](https://github.com/lightninglabs/taproot-assets/pull/716/files). We could probably come up with some fancy collectors to retrieve everything.

Given that the tapd instances that the loadtests run against are long-running, the metrics will persist across multiple loadtest runs.

Reply:
So next steps:
- Strip any redundant changes from this PR: Let's keep the push metrics that are introduced here only for tracking the duration of the test
- Enhance tapd itself with any type of collector we deem appropriate (gRPC, proof related etc): Decorate our grafana dashboards with the prom metrics of the long-running alice & bob nodes

Tracking extra things in this layer (`itest/loadtest`) is kind of useless or very limiting as this is the client side and we don't really care about this code area, or it would be much harder extracting the tapd metrics on this level and then pushing it to prom